packages/jazz-vue/src/createJazzVueApp.ts

```diff
@@ -0,0 +1,342 @@
+    BrowserGuestContext,
+    consumeInviteLinkFromWindowLocation,
+} from "jazz-browser";
+import {
+    Account,
+    AuthMethod,
+    CoValue,
+    CoValueClass,
+    ID,
+    DepthsIn,
+    DeeplyLoaded,
+    AnonymousJazzAgent,
+    subscribeToCoValue,
+} from "jazz-tools";
+
+export interface JazzVueApp<Acc extends Account> {
+    JazzProvider: Component;
+
+    useAccount(): { me: ComputedRef<Acc>; logOut: () => void };
+    useAccount<D extends DepthsIn<Acc>>(
+        depth: D,
+    ): {
+        me: ComputedRef<DeeplyLoaded<Acc, D> | undefined>;
+        logOut: () => void;
+    };
+
+    useAccountOrGuest(): { me: ComputedRef<Acc | AnonymousJazzAgent> };
+    useAccountOrGuest<D extends DepthsIn<Acc>>(
+        depth: D,
+    ): {
+        me: ComputedRef<DeeplyLoaded<Acc, D> | undefined | AnonymousJazzAgent>;
+    };
+
+    useCoState<V extends CoValue, D>(
+        Schema: CoValueClass<V>,
+        id: ID<V> | undefined,
+        depth?: D & DepthsIn<V>,
+    ): Ref<DeeplyLoaded<V, D> | undefined>;
+
+    useAcceptInvite<V extends CoValue>(args: {
+        invitedObjectSchema: CoValueClass<V>;
+        onAccept: (projectID: ID<V>) => void;
+        forValueHint?: string;
+    }): void;
+}
+
+const JazzContextSymbol = Symbol("JazzContext");
+
+export function createJazzVueApp<Acc extends Account>({
+    AccountSchema = Account as any,
+} = {}): JazzVueApp<Acc> {
+    const JazzProvider = defineComponent({
+        name: "JazzProvider",
+        props: {
+            auth: {
+                type: [String, Object] as PropType<AuthMethod | "guest">,
+                required: true,
+            },
+            peer: {
+                type: String as PropType<`wss://${string}` | `ws://${string}`>,
+                required: true,
+            },
+            storage: {
+                type: String as PropType<"indexedDB" | "singleTabOPFS">,
+                default: undefined,
+            },
+        },
+        setup(props, { slots }) {
+            const ctx = ref<
+                BrowserContext<Acc> | BrowserGuestContext | undefined
+            >(undefined);
+
+            provide(JazzContextSymbol, ctx);
+
+            onMounted(async () => {
+                try {
+                    console.log("props.auth", props.auth);
+                    const context = await createJazzBrowserContext<Acc>(
+                        props.auth === "guest"
+                            ? { peer: props.peer, storage: props.storage }
+                            : {
+                                  AccountSchema,
+                                  auth: props.auth,
+                                  peer: props.peer,
+                                  storage: props.storage,
+                              },
+                    );
+
+                    ctx.value = {
+                        ...context,
+                        logOut: () => {
+                            context.logOut();
+                            ctx.value = undefined;
```

Comment:
**observation:** Not sure if we need to create a new context on logout

@aeplay How do we handle the login -> logout -> login flow?

Reply:
Fixed